js-plot/README.md

```diff
@@ -193,25 +314,217 @@ TODO explain vega-spec.json layer construction
+Vega-Lite can superimpose one chart on top of another with [layers](https://vega.github.io/vega-lite/docs/layer.html) keyword.
+Let's construct each layer separately and super impose them at the end.
+
+The 2x^3 - 4x^2 + 6 equation is plotted by using a [sequence generator](https://vega.github.io/vega-lite/docs/data.html#sequence) and a [formula transform](https://vega.github.io/vega-lite/docs/calculate.html).
+
+```js
+const equation_line = {
+  "data": {"sequence": {"start": -4, "stop": 4, "step": 0.1, "as": "x"}},
+  "transform": [
+    {
+      "calculate": "2 * datum.x * datum.x * datum.x - 4 * datum.x * datum.x + 6",
+      "as": "y"
+    }
+  ],
+  "mark": "line",
+  "encoding": {
+    "x": {"type": "quantitative", "field": "x"},
+    "y": {"field": "y", "type": "quantitative"},
+    "color": {"value": "#f00"}
+  }
+};
+```
+
+To show where the root is we draw a dotted vertical line at x is -1 with an title. For the ruler and title we reuse the data via layers and use a [rule marker](https://vega.github.io/vega-lite/docs/rule.html) and [text marker](https://vega.github.io/vega-lite/docs/text.html) respectivly.
+
+```js
+const root_rule = {
+  "data": {"values": [{"x": -1}]},
+  "encoding": {"x": {"field": "x", "type": "quantitative"}},
+  "layer": [
+    {"mark": {"type": "rule", "strokeDash": [4, 8]}},
+    {"mark": {"type": "text", "align": "left", "dx": 4, "text": "Root"}}
+  ]
+};
+```
+
+When we plot the x and y of each iteration we can no longer see the order of iterations.
+So we will use a text marker above each circle to tell us to which iteration it belongs.
+
+```js
+const iterations_scatter = {
+  "data": {
+    "values": iterations
+  },
+  "encoding": {
+    "x": {"field": "x", "type": "quantitative"},
+    "y": {"field": "y", "type": "quantitative"},
+    "text": {"field": "index", "type": "nominal"}
+  },
+  "layer": [
+    {
+      "mark": {"type": "circle", "tooltip": {"content": "data"}},
+      "selection": {"grid": {"type": "interval", "bind": "scales"}}
+    },
+    {"mark": {"type": "text", "dy": -10}}
+  ]
+};
+```
+
+Superimpose the equation, root ruler and iteration plot into a single visualization with
+
+```js
+const spec = {
+  "$schema": "https://vega.github.io/schema/vega-lite/v4.json",
+  "width": 800,
+  "height": 600,
+  "layer": [
+    equation_line,
+    root_rule,
+    iterations_scatter
+  ]
+};
+```
+
+The HTML page with all JavaScript put together to make a composite plot is
+
+```html
+<html>
+
+<head>
+  <script type="text/javascript" src="newtonraphson.js"></script>
+  <script src="https://cdn.jsdelivr.net/npm/vega@5.13.0"></script>
+  <script src="https://cdn.jsdelivr.net/npm/vega-lite@4.13.0"></script>
+  <script src="https://cdn.jsdelivr.net/npm/vega-embed@6.8.0"></script>
+</head>
+
+<body>
+  <div id="plot"></div>
+  <script>
+    createModule().then((rootfinding) => {
+      const initial_guess = -3;
```

Comment:
```suggestion
      const initial_guess = -4;
```
(If we're going with that init value)

Reply:
thanks I missed that on during https://github.com/NLESC-JCER/run-cpp-on-web/pull/37/commits/1ed07a6d9974b6292619a93bcf5edb74c6f72beb